oneview/resource_logical_enclosure.go

```diff
@@ -0,0 +1,377 @@
+							Optional: true,
+						},
+					},
+				},
+			},
+			"enclosure_group_uri": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"enclosure_uris": {
+				Required: true,
+				Type:     schema.TypeList,
+				Elem:     &schema.Schema{Type: schema.TypeString},
+			},
+			"firmware": {
+				Optional: true,
+				Type:     schema.TypeSet,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"firmware_baseline_uri": {
+							Type:     schema.TypeString,
+							Optional: true,
+						},
+						"firmware_update_on": {
+							Type:     schema.TypeString,
+							Optional: true,
+						},
+						"force_install_firmware": {
+							Type:     schema.TypeBool,
+							Optional: true,
+						},
+						"logical_interconnect_update_mode": {
+							Type:     schema.TypeString,
+							Optional: true,
+						},
+						"update_firmware_on_unmanaged_interconnect": {
+							Type:     schema.TypeBool,
+							Optional: true,
+						},
+						"validate_if_li_firmware_update_is_non_disruptive": {
+							Type:     schema.TypeBool,
+							Optional: true,
+						},
+					},
+				},
+			},
+
+			"ip_addressing_mode": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"ipv4_range": {
+				Optional: true,
+				Type:     schema.TypeSet,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"dns_servers": {
+							Type:     schema.TypeString,
+							Optional: true,
+						},
+						"domain": {
+							Type:     schema.TypeString,
+							Optional: true,
+						},
+						"gateway": {
+							Type:     schema.TypeString,
+							Optional: true,
+						},
+						"ip_range_uri": {
+							Type:     schema.TypeString,
+							Optional: true,
+						},
+						"name": {
+							Type:     schema.TypeString,
+							Optional: true,
+						},
+						"subnet_mask": {
+							Type:     schema.TypeString,
+							Optional: true,
+						},
+					},
+				},
+			},
+			"logical_interconnect_uris": {
+				Optional: true,
+				Type:     schema.TypeSet,
+				Elem:     &schema.Schema{Type: schema.TypeString},
+				Set:      schema.HashString,
+			},
+			"name": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"power_mode": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"scaling_state": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"scopes_uri": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"status": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"type": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"uri": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+		},
+	}
+
+}
+func resourceLogicalEnclosureCreate(d *schema.ResourceData, meta interface{}) error {
+	config := meta.(*Config)
+	logicalEnclosure := ov.LogicalEnclosure{
+		Name:              d.Get("name").(string),
+		ScopesUri:         utils.NewNstring(d.Get("scopes_uri").(string)),
```

Comment:
ScopesUri is not sent during POST. Instead InitialScopeUris field can be sent during POST.

Reply:
GO structure has only ScopeUri , it is not matching with terraform. Since it is not mandatory, removed ScopeUri .